clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/ElasticSearchEntityTagger.java

```diff
@@ -202,34 +197,73 @@ private void upsertEntityTags(String type,
+  }
 
+  private void upsertEntityTags(String name,
+                                String cloudProvider,
+                                String accountId,
+                                String region,
+                                String namespace,
+                                String entityType,
+                                String entityId,
+                                Object tagValue,
+                                Long timestamp) {
+    upsertEntityTags(
+      upsertEntityTagsDescription(name, cloudProvider, accountId, region, namespace, null, entityType, entityId, tagValue, timestamp));
+  }
+
+  private void upsertEntityTags(UpsertEntityTagsDescription description) {
+    AtomicOperation<?> operation = upsertEntityTagsAtomicOperationConverter.buildOperation(description);
+
+    operate(operation);
+  }
+
+  private void operate(AtomicOperation<?> operation) {
+    Task originalTask = TaskRepository.threadLocalTask.get();
     try {
-      TaskRepository.threadLocalTask.set(new DefaultTask(this.getClass().getSimpleName()));
-      upsertEntityTagsAtomicOperation.operate(Collections.emptyList());
+      TaskRepository.threadLocalTask.set(
+        Optional.ofNullable(originalTask).orElse(new DefaultTask(ElasticSearchEntityTagger.class.getSimpleName()))
+      );
+      run(operation);
     } finally {
-      TaskRepository.threadLocalTask.set(null);
+      TaskRepository.threadLocalTask.set(originalTask);
     }
   }
 
-  private static UpsertEntityTagsDescription upsertEntityTagsDescription(String type,
-                                                                         String prefix,
+  @VisibleForTesting
+  protected void run(AtomicOperation<?> operation) {
+    operation.operate(Collections.emptyList());
+  }
+
+  private static UpsertEntityTagsDescription upsertCategorizedEntityTagsDescription(String type,
+                                                                                    String prefix,
+                                                                                    String cloudProvider,
+                                                                                    String accountId,
+                                                                                    String region,
+                                                                                    String category,
+                                                                                    String entityType,
+                                                                                    String entityId,
+                                                                                    String key,
+                                                                                    String value,
+                                                                                    Long timestamp) {
+    String name = prefix + key;
```

Comment:
the slightly weird concatenation used for alerts/notices has always struck me a little weird ... let's keep an eye out and prevent this sort of pattern from happening anywhere else.

```
{
        "name": "spinnaker_ui_alert:astrid_rules:jar:netflix.base-server<4.5.47",
        "namespace": "astrid_rules",
        "category": "deprecation",
        "value": {
          "imageId": "....",
          "tagline": "*Brought to you by [Astrid](http://......)*",
          "type": "alert",
          "title": "**netflix.base-server <4.5.47......*",
          "message": "Minimize support burden. Keep systems up-to-date. Upgrade to latest version. "
        },
        "valueType": "object"
      },
```

The `name` and `value.type` seem to indicate the same thing and argument could be made that the name doesn't actually even need a `spinnaker_ui_alert` prefix.

Reply:
I won't be adding it anywhere else, but I also don't want to change it now that it's already been shipped out like this.